afs/afs-core/src/test/java/com/powsybl/afs/LocalTaskMonitorTest.java

```diff
@@ -91,11 +97,36 @@ public void onEvent(TaskEvent event) {
             } catch (IllegalArgumentException ignored) {
             }
 
+            monitor.updateTaskCancelableFuture(task.getId(), null);
+            assertEquals(1, events.size());
+            assertEquals(new TaskCancelableStatusChangeEvent(task.getId(), 3L, false), events.pop());
+            AtomicInteger taskCounter = new AtomicInteger(0);
+            AtomicBoolean interrupted = new AtomicBoolean(false);
+            CompletableFuture<Integer> dummyTaskProcess = CompletableFuture.supplyAsync(() -> {
+                for (int i = 0; i < 1000; i++) {
+                    taskCounter.incrementAndGet();
+                    try {
+                        Thread.sleep(100);
```

Comment:
Maybe we could simply use a `TaskCounter` or a `CompletableFuture` to wait until it gets interrupted ? (see for example `CompletableFutureTaskTest#whenCancelDuringExecutionThenThrowAndInterrupt`)
The test relies on an arbitrary value which may not be necessary.

Not blocking though.